src/corelibs/U2View/src/ov_msa/ColorSchemaDialogController.cpp

```diff
@@ -374,6 +379,10 @@ void ColorSchemaSettingsPageWidget::sl_schemaChanged(int index) {
 void ColorSchemaSettingsPageWidget::sl_onColorsDirButton() {
     QString path = colorsDirEdit->text();
     QString dir = U2FileDialog::getExistingDirectory(this, tr("Choose Folder"), path, QFileDialog::ShowDirsOnly | QFileDialog::DontResolveSymlinks);
+    if (!FileAndDirectoryUtils::isDirectoryWritable(dir)) {
```

Comment:
You already have an existing dir path: all your need is to check if it is writable FileInfo(path).isWritable()
This way you can avoid the creation of a 1-line helper methods.